gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DuplicateObservationStatementMapper.java

```diff
@@ -66,37 +64,72 @@ private static List<String> getLinksetsIn(RCMRMT030101UKEhrComposition ehrCompos
                 .toList();
     }
 
-    private static void findAndMergeNonTruncatedObservationStatementIntoTruncatedObservationStatement(
-            RCMRMT030101UKObservationStatement truncatedObservationStatement, List<RCMRMT030101UKComponent4> components) {
-
-
-        RCMRMT030101UKAnnotation truncatedPertinentAnnotation = getPertinentAnnotation(truncatedObservationStatement);
-        String truncatedAnnotation = truncatedPertinentAnnotation.getText();
-        String annotationPrefix = truncatedAnnotation.substring(0, truncatedAnnotation.length() - CHAR_LIMIT_FOR_TRUNCATION);
-
-        for (Iterator<RCMRMT030101UKComponent4> observationIterator = components.iterator(); observationIterator.hasNext();) {
-            var observationStatement = observationIterator.next().getObservationStatement();
-            if (observationStatement != null
-                    && !areSameObservationStatements(truncatedObservationStatement, observationStatement)
-                    && hasSinglePertinentInformation(observationStatement)
-                    && doesTruncatedAnnotationMatchOtherAnnotation(truncatedPertinentAnnotation,
-                          getPertinentAnnotation(observationStatement))
-                    && observationsAreCodedTheSame(truncatedObservationStatement, observationStatement)
-                    && areAdditionalFieldsEmpty(observationStatement)) {
-                observationIterator.remove();
-                truncatedPertinentAnnotation.setText(annotationPrefix + getPertinentAnnotation(observationStatement).getText());
-                LOGGER.info(
-                        "ObservationStatement: '{}' Is truncated version of '{}' and will be merged into a single observation.",
-                        truncatedObservationStatement.getId().getRoot(),
-                        observationStatement.getId().getRoot()
-                );
+    private static void findAndMergeOrRemoveDuplicates(RCMRMT030101UKObservationStatement observationStatementWithPertinentInfo,
+                                                       List<RCMRMT030101UKComponent4> components) {
+        if (isAnnotationTruncated(observationStatementWithPertinentInfo)) {
+            mergeIfTruncated(observationStatementWithPertinentInfo, components);
+        } else {
+            removeExactDuplicates(observationStatementWithPertinentInfo, components);
```

Comment:
Are there any unit tests within `DuplicateObservationStatementMapperTest` which tests this code?

#TDD4Lyfe